crates/uv/src/commands/project/mod.rs

```diff
@@ -1119,6 +1120,24 @@ impl WorkspacePython {
         project_dir: &Path,
         no_config: bool,
     ) -> Result<Self, ProjectError> {
+        // Helper to parse a concrete version from the request (non-range) for compatibility check
+        fn pep440_version_from_request(request: &PythonRequest) -> Option<uv_pep440::Version> {
+            let version_request = match request {
+                PythonRequest::Version(version)
+                | PythonRequest::ImplementationVersion(_, version) => version,
+                PythonRequest::Key(download_request) => download_request.version()?,
+                _ => {
+                    return None;
+                }
+            };
+            if matches!(version_request, uv_python::VersionRequest::Range(_, _)) {
+                return None;
+            }
+            // Remove Python variant and parse to PEP 440 version
+            let s = version_request.clone().without_python_variant().to_string();
+            Some(uv_pep440::Version::from_str(&s).unwrap())
```

Comment:
Do we need to roundtrip to a string and parse? Can't we just construct a `Version` from the u8s? I'd probably make this a `VersionRequest::as_version` utility too.